source/CCustomOpcodeSystem.cpp

```diff
@@ -1555,7 +1562,8 @@ namespace CLEO
 	{
 		char gxtBuff[8]; // 7 + terminator character
 		auto gxt = OPCODE_READ_PARAM_STRING_BUFF(gxtBuff, sizeof(gxtBuff));
-		auto txt = OPCODE_READ_PARAM_STRING();
+		char buf[MAX_STR_LEN];

```

Comment:
As visible `buf` variable do not introduce anything meaningful to the opcode implementation. It also starts polluting scope with multiple buffers.

OPCODE_READ_PARAM_STRING_BUFF should return nothing and be used as already existing 
```
char text[MAX_STR_LEN]; OPCODE_READ_FORMATTED_STRING(thread, text, sizeof(text), format);
```

Reply:
Can you create a unique variable in macro? I thought it's impossible.

Will it work like this:

```
auto str1 = OPCODE_PARAM_READ_STRING();
auto str2 = OPCODE_PARAM_READ_STRING();
auto str3 = OPCODE_PARAM_READ_STRING();
```

where each str gets it's own buffer?

Reply:
Tried but no idea. It need to be something like
```
OPCODE_PARAM_READ_STRING(name);
```
It would make sense. Then it will be possible to have overload version with limited size:
```
OPCODE_PARAM_READ_STRING(name, 4);
```
Which also can benefit from not copying the data if not necessary.